libraries/chain/include/eosio/chain/exceptions.hpp

```diff
@@ -276,6 +276,10 @@ namespace eosio { namespace chain {
                                     3040017, "Transaction includes disallowed extensions (invalid block)" )
       FC_DECLARE_DERIVED_EXCEPTION( tx_resource_exhaustion, transaction_exception,
                                     3040018, "Transaction exceeded transient resource limit" )
+      FC_DECLARE_DERIVED_EXCEPTION( ro_trx_temporary, transaction_exception,
+                                    3040019, "Read-only transaction temporary failure" )
+      FC_DECLARE_DERIVED_EXCEPTION( ro_trx_permanent, transaction_exception,
+                                    3040020, "Read-only transaction permanent failure" )
```

Comment:
Seems like these should go under `resource_exhausted_exception` instead.
Also I think they could have a better description, maybe something like:
`Read-only transaction eos-vm-oc compile [temporary|permanent] failure.`
Also maybe call them `_ro_trx_vm_oc_compile_temporary` or something like that.

Reply:
I was thinking of generic exceptions for read-only transactions. But it is better to be more specific.